### PR TITLE
States/fe/random fill

### LINKED ISSUE #214: Refactor random fill to handle new multiple options changes
Besides the act itself of generating random choices based on the available options, this issue is also aimed to the lock / unlock funcionality of an option, since that is part of the random fill.

Basically, if a course (disciplina) is locked, the `<RandomFill>` component ignores it and does not generate a random choice to that one.

In essence, in order to finish this issue, it should not be enough to just allow random courses to be picked when the user clicks the button. It should also handle lock / unlock logic. 

Currently, the padlocks near the dropdowns to select a class option for a course are not correctly working and that is also part of the issue.

### LINKED ISSUE #260: Copy an empty schedule option and pasting it
Copying an empty schedule and pasting it creates an empty selector. Also, it thinks there is some class checked and the modal has some bugs because of that.

Fix the bug somewhere in the paste function.

### LINKED ISSUE #261: Classes types filter
The filters below the schedule stop working since the refactor

### PR DESCRIPTION
**Pull request by Wallen-Ribeiro**
Closes #214, Closes #261, Closes #260 

Debugging checklist:

- [x] Random classes update when classes change
- [x] Weird bug when we click on the first course option, it doesn't update the states as it should
- [x] Locks not option individual
- [ ] With a big number of courses, takes time

### CODE REVIEW

**Comment by thePeras** [comment]
Notice that with a lot of picked courses it takes time and in some cases crashes the browser. It is not a usual case, but we need to rethink the Cartesian approach in the future.

I make this PR also closing #260 and #261

File: src/utils/index.ts [hunk 1]
```diff
@@ -174,7 +174,7 @@ const getClassTypeClassName = (type: string) => {
   }
 }
 
-const getClassType = (type: string) => {
+const getClassType = (type: String) => {
```
Comment (tomaspalma) [inline src/utils/index.ts:177]: Why the change from `string` to `String`? According to [this](https://www.typescriptlang.org/docs/handbook/literal-types.html#string-literal-types) and [this](https://stackoverflow.com/questions/14727044/what-is-the-difference-between-types-string-and-string), shouldn't it be `string`?

**Review by tomaspalma — APPROVED** [review]
:rocket:
